convention.md

```diff
@@ -18,10 +18,9 @@ The underscore (`_`) is used and reserved for Homie *node arrays*.
 
 ### Payload
 
-Every MQTT message payload MUST be sent as string.
-If a payload is of a numeric data type, it MUST be converted to string.
+Every MQTT message payload MUST be sent as UTF-8 encoded string.
+A numeric data type MUST contain a numeric value published as string with the decimal separator, if any, being a dot.
```

Comment:
Reads as a bit confusing.

```suggestion
A numeric data type MUST be published as a string with a dot as the decimal separator, e.g. "123.456".
```

Reply:
The important point for @boc-tothefuture is that an integer payload can never an empty string, that is somehow lost in your sentence. And the decimal separator is not a requirement for numerical values, only if there is one, it should be the dot.

Reply:
The first should be implicit (because "" is not a valid numeric !?). If we want to emphasize this I'm sorry to tell you that I didn't understand this from your version.
Regarding the second: Imho a decimal delimiter is always only needed if decimal places are present. If you want to specifically state that maybe it would be best to divide the whole sentence in two.

Reply:
Actually I was happy with the original sentence (I just noticed that we don't specify the decimal separator), but it was found to be inaccurate. @boc-tothefuture could you please present your argumentation here?

Reply:
It is my opinion that we should specify that empty values, besides for string type are not allowed. I could see an implementation thinking that sending an empty string for a property payload _clears_ the value.  I don't think that, but I could see how someone might.

For example, is an empty string OK for type enum?  The convention doesn't say that it is not, but I also think we can't actually create it because the way that we escape commas in the parsing of enum definitions.

For the comment that "" is not a valid numeric.  I agree, but some languages don't.  For example in ruby `"".to_i` happily returns 0.  If someone had only a ruby background, they might think an empty payload is acceptable to represent zero for a numeric.

This is beyond the scope of this item pull request, but I was going to put together a PR to more clearly define the acceptable values. 

As for this particular PR, I think defining the dot as the decimal separator is good.

Reply:
Empty messages on retained topics delete the retained topic. I did not check if an empty string is equal to a truly empty message.

The convention should disallow empty values - however not just for numbers. The change in this PR should clarify that each property can only be set within the range of its defined datatype. For numbers this is at least one digit, for enums this is exactly 1-of-n and for strings this is any combination of 0..* characters (incl. empty string).

In #51 it might be a good idea to emphasize this requirement.

As for this PR, @davidgraeff let me know if you would like to see a new suggestion from me.

Reply:
@ThomDietrich I definitely welcome another suggestion.

Reply:
```suggestion
A numeric data type MUST be converted to a literal string with the dot character as the decimal separator for the `float` data type.
An empty message is not valid as payload for numeric data types.
The value published as payload MUST be valid for the respective property/attribute type as per the convention definition below.
```